### ISSUE TITLE
What are the new stuffs (concepts, algorithms, tools) that do you want to learn?

### ISSUE THREAD

**Issue by seagatesoft** [ariya/ama#29]
Also, how is your regular approach to learn something?

**Comment by ariya** [comment]
There are tons of new things I want to learn! However, realistically speaking my body and my mental power have limitations, and thus I also learn not to chew more that I could.

My optimal approach to learn something new to **immerse myself**. There were times during my graduate research (see #21) where I needed to master various new concepts. For instance, when I had to learn [quaternions](https://en.wikipedia.org/wiki/Quaternion) from scratch, for weeks and weeks I scribbled various quaternions operations everywhere, used any idle moments to think about it, kept that Hamilton book next to me bed, implemented tons of example computations in Matlab and C++, and so on and so forth. Back then, I often joked that if someone wakes me up at the middle of the night, my subject of discussion would be quaternions.

Now, back to the main question. What do I want to learn in the near future?

Up to this point, I always call myself a Lua beginner. While I can defend myself using a few Lua tricks I know, I still never worked on _serious_ projects using Lua. This year, my target is to **learn Lua** to a point where I can comfortably say that it reaches the same level of understanding of other languages that I know.

Thank you @seagatesoft!
